management-service/src/main/java/pl/pwr/zpi/notifications/email/service/EmailReceiverService.java

```diff
@@ -15,32 +17,60 @@
 public class EmailReceiverService {
 
     private final EmailRepository emailRepository;
+    private final EmailValidator emailValidator = EmailValidator.getInstance();
 
     public List<EmailReceiver> getAllEmails() {
         return emailRepository.findAll();
     }
 
     public void addNewEmail(EmailReceiverDTO emailReceiver) {
+        long now = System.currentTimeMillis();
+
         checkIfEmailExists(emailReceiver);
         EmailReceiver receiver = EmailReceiver.builder()
                 .receiverName(emailReceiver.getName())
                 .receiverEmail(emailReceiver.getEmail())
-                .createdAt(System.currentTimeMillis())
+                .createdAt(now)
+                .updatedAt(now)
                 .build();
         emailRepository.save(receiver);
     }
 
-    public EmailReceiver updateEmail(Long id, EmailReceiverDTO emailReceiver) {
+    public EmailReceiver updateEmail(Long id, EmailReceiverUpdateRequest updateRequest) {
         var receiver = getEmailReceiver(id);
 
-        checkIfUserCanUpdateEmail(emailReceiver.getEmail(), id);
+        checkIfUserCanUpdateEmail(updateRequest.email(), id);
+        patchEmail(receiver, updateRequest);
 
-        receiver.setReceiverName(emailReceiver.getName());
-        receiver.setReceiverEmail(emailReceiver.getEmail());
-        receiver.setUpdatedAt(System.currentTimeMillis());
         return emailRepository.save(receiver);
     }
 
+    private void patchEmail(EmailReceiver emailReceiver, EmailReceiverUpdateRequest updateRequest) {
+        if(updateRequest.email() != null) {
+            validateEmail(updateRequest.email());
+            emailReceiver.setReceiverEmail(updateRequest.email());
+        }
+
+        if(updateRequest.name() != null) {
+            validateReceiverName(updateRequest.name());
+            emailReceiver.setReceiverName(updateRequest.name());
+        }
+
+        emailReceiver.setUpdatedAt(System.currentTimeMillis());
+    }
+
+    private void validateEmail(String email) {
+        if(!emailValidator.isValid(email)) {
+            throw new RuntimeException("Invalid email");
+        }
+    }
+
+    private void validateReceiverName(String name) {
+        if(name.isEmpty()) {
+           throw new RuntimeException("Receiver name is empty");
+        }
+    }
```

Comment:
again you can validate this with annotation:
@Email(message = "Email is not valid", regexp = "^[a-zA-Z0-9_!#$%&'*+/=?`{|}~^.-]+@[a-zA-Z0-9.-]+$")
@NotEmpty(message = "Email cannot be empty")
private String email;

Reply:
As explained before